samza/sso-account-updater/src/main/java/org/sunbird/jobs/samza/service/SSOAccountUpdaterService.java

```diff
@@ -0,0 +1,183 @@
+            passedRoles.add(SSOAccountUpdaterParams.PUBLIC.name());
+        }
+
+        String passedOrgId = (String) passedOrg.get(SSOAccountUpdaterParams.identifier.name());
+        if (CollectionUtils.isEmpty(organisations)) {
+            updateMap.put(SSOAccountUpdaterParams.organisations.name(), getOrganisationsListFromPayloadData(passedOrgId, userId, passedRoles));
+        } else {
+            boolean isUserUpdateRequired = true;
+            passedRoles.remove(SSOAccountUpdaterParams.PUBLIC.name());
+
+            for (Map<String, Object> organisation : organisations) {
+                String linkedOrgId = (String) organisation.get(SSOAccountUpdaterParams.organisationId.name());
+                //find linked org
+                if (StringUtils.equalsIgnoreCase(linkedOrgId, passedOrgId)) {
+                    //match roles
+                    List<String> assignedRoles = (List<String>) organisation.get(SSOAccountUpdaterParams.roles.name());
+                    assignedRoles.remove(SSOAccountUpdaterParams.PUBLIC.name());
+                    if (CollectionUtils.isEqualCollection(passedRoles, assignedRoles)) {
+                        isUserUpdateRequired = false;
+                        break;
+                    }
+                }
+            }
+
+            if (isUserUpdateRequired) {
+                updateMap.put(SSOAccountUpdaterParams.organisations.name(), getOrganisationsListFromPayloadData(passedOrgId, userId, passedRoles));
+            }
+        }
+
+        if (MapUtils.isNotEmpty(updateMap)) {
+            updateMap.put(SSOAccountUpdaterParams.userId.name(), userId);
+
+            Map<String, Object> updateUserRequest = new HashMap<>();
+            updateUserRequest.put(SSOAccountUpdaterParams.request.name(), updateMap);
+
+            updateUser(updateUserRequest);
+            Logger.info("SSOAccountUpdaterService:processMessage: User data updated in the system for userId - " + userId);
+        }
+
+    }
+
+    private List<Map<String, Object>> getOrganisationsListFromPayloadData(String passedOrgId, String userId, List<String> passedRoles) {
+        Map<String, Object> organisationMap = new HashMap<>();
+        organisationMap.put(SSOAccountUpdaterParams.organisationId.name(), passedOrgId);
+        organisationMap.put(SSOAccountUpdaterParams.userId.name(), userId);
+        organisationMap.put(SSOAccountUpdaterParams.roles.name(), passedRoles);
+
+        List<Map<String, Object>> linkedOrgList = new ArrayList<>();
+        linkedOrgList.add(organisationMap);
+
+        return linkedOrgList;
+    }
+
+    private Map<String, Object> getOrg(String channel, String orgExternalId) throws Exception {
+
+        Map<String, String> filterMap = new HashMap<>();
+        filterMap.put(SSOAccountUpdaterParams.channel.name(), channel);
+        filterMap.put(SSOAccountUpdaterParams.externalId.name(), orgExternalId);
+
+        Map<String, Object> requestMap = new HashMap<>();
```

Comment:
Can't an elastic search util common method be used to perform the search query and return response to avoid code duplications?